### ISSUE TITLE
Epic: Implement remaining JetBrains IDE's (CLion, WebStorm, Rider, RubyMine)

### ISSUE THREAD

**Issue by loujaybee** [gitpod-io/gitpod#7751]
We are currently only supporting the following JetBrains IDE's
- [x] GoLand
- [x] IntelliJ
- [x] PHPStorm
- [x] PyCharm

However, we need to add remote development backends support for: 
- [x] CLion (https://www.gitpod.io/blog/gitpod-jetbrains)
- [x] Rider (https://www.gitpod.io/blog/gitpod-jetbrains)
- [x] RubyMine (https://www.gitpod.io/blog/webstorm-rubymine-now-in-beta)
- [x] WebStorm (https://www.gitpod.io/blog/webstorm-rubymine-now-in-beta)

Related:
- https://github.com/gitpod-io/gitpod/pull/14524

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_3p1my)

**Callout:** 👋 Are you interested in this feature? We would be interested to [talk to you to hear your feedback](https://calendly.com/lou-gitpod/product-feedback).

Internal conversation [[1](https://gitpod.slack.com/archives/C02BRJLGPGF/p1642760813079100)]

**Comment by akosyakov** [comment]
I think it depends on https://github.com/gitpod-io/gitpod/issues/7023 Till we cannot do it automatically with smoke testing it will be tedious to review so many PRs.

**Comment by sinansonmez** [comment]
Is there any expected time of arrival for other IDEs support?

**Comment by loujaybee** [comment]
> Is there any expected time of arrival for other IDEs support?

Gitpod current focus is on the IntelliJ experience (which has a lot of overlap with the other JetBrains IDEs), which is also where JetBrains themselves are focussing a lot of their remote development energy. As the experience improves for the other IDEs we'll be looking to roll those as soon as we reasonably can 🙏

**Comment by sualehasif** [comment]
Waiting for the CLion support :)

**Comment by akosyakov** [comment]
With https://github.com/gitpod-io/leeway/issues/93 done, we can simplify our build process to add more components easier.

**Comment by CoolCoderSuper** [comment]
Can't wait for Rider

**Comment by loujaybee** [comment]
Hey folks, wanted to give a quick update (as many have asked!). Firstly, thanks so much to all our users who have given [feedback in the community](https://www.gitpod.io/community) for our JetBrains integration. Please continue to do so, we're learning a lot from you and we're able to refine the experience based on your input. 

If you're late to the party, I wanted to give a little background and context into Gitpod IDE support and why we've not yet released full support of the IDEs just yet. For background, [we have a partnership with JetBrains](https://www.gitpod.io/blog/gitpod-jetbrains). In real terms is that we have conversations with the friendly JetBrains people to help input on the future of JetBrains Gateway as a whole and help create a really great remote dev experience with JetBrains in Gitpod and beyond! 

Opening up an IDE is only one step in the process to bringing a great developer experience in Gitpod. Whilst a lot of the Gitpod functionality is shared across the JetBrains IDEs, such as [prebuild support](https://github.com/gitpod-io/gitpod/issues/6740), [integration of Gitpod task terminals](https://github.com/gitpod-io/gitpod/issues/10579), port forwarding and [port behaviours](https://github.com/gitpod-io/gitpod/issues/10581). There are also lots of language specific product changes that need to be made in both Gitpod and JetBrains. 

**For example:** 
* Ensuring coverage of plugins / VS Code extensions for the language - [see recently added Oracle support](https://open-vsx.org/extension/Oracle/oracledevtools) in OpenVSX
* Our documentation support is good - [see recently updated Java page](https://www.gitpod.io/docs/introduction/languages/java)
* Integration with build tools for the language - we are [working on Maven right now](https://github.com/gitpod-io/gitpod/issues/6740), etc). 

For that reason we've been a bit deliberate in how (and when) we roll out other IDE support. We don't want to open up an IDE for our Gitpod users to go try the integration, but then be dissapointed (let's say if some aspect of the integration is not up to the standard they expect, some plugin doesn't work, etc). 

**In simpler terms: The quality of the experience is really important to us.** 

We just opened up WebStorm and RubyMine (yet to be announced). So hopefully we will get to the remaining IDEs in the coming weeks !  Come join us on Discord, chat about your experience and give us your feedback (we are listening!). We're super excited for what's to come with JetBrains + Gitpod 🙏

**Comment by loujaybee** [comment]
* ~RubyMine~
* ~WebStorm~

https://www.gitpod.io/blog/webstorm-rubymine-now-in-beta

**Comment by andreafalzetti** [comment]
- Rider + CLion docs published: 
- Changelog in draft, will be published soon

Closing as complete ✅
